cpp/data_structures/dynamic_tree/DynamicTree.hpp

```diff
@@ -0,0 +1,441 @@
+﻿#include <iostream>
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#pragma once
+
+#ifndef DYNAMIC_TREE_H
+#define DYNAMIC_TREE_H
+
+namespace cpstl {
+template <class T>
+class DynamicTree;
+
+// ===================
+// === Nodes class ===
+// ===================
```

Comment:
If you need to include some things in the comment, please use the C++ convention like

/**
 * Dynamic Tree Node, an internal representation of a tree node
**/